### PR TITLE
Add episode 16

### PR DESCRIPTION
**Pull request by bitsondatadev**
Let's not merge until Monday (5-3).

### CODE REVIEW

File: _episodes/16.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+     time: 2015
+   - title: "Demo"
+     desc: "Using the Druid Web UI to create an ingestion spec querying via Trino"
+     time: 3689
+   - title: "Question of the week"
+     desc: "Does the Druid connector?"
+     time: 4220
+---
+
+Trino nation, We want to hear from you! If you have a question or pull request
```
Comment (mosabua) [inline _episodes/16.md:23]: , we

File: _episodes/16.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+ ([@samarthjain11](https://www.linkedin.com/in/samarthjain11/))
+ * Parth Brahmbhatt, Senior Software Engineer at Netflix 
+ ([@brahmbhattparth](https://twitter.com/brahmbhattparth/))
+ * Rachel Pedreschi, VP Community and Developer Relations at 
+ [Imply](https://imply.io/) ([@rachelpedreschi](https://twitter.com/rachelpedreschi))
+
+## Release 356
+
+Release notes discussed: <https://trino.io/docs/current/release/release-356.html>
+
```
Comment (mosabua) [inline _episodes/16.md:55]: Add link to https://github.com/trinodb/trino/pull/7757

Comment (bitsondatadev) [reply _episodes/16.md:55]: I was assuming we wouldn't merge this until 356 was merged..which I assumed would be soon

File: _episodes/16.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+ - General:
+   - MATCH_RECOGNIZE - query over range of record with trending and such
```
Comment (mosabua) [inline _episodes/16.md:57]: - MATCH_RECOGNIZE clause support, used to detect patterns in a set of rows within a single query

File: _episodes/16.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+   - soundex function
+   - Property limit planning time (and improved behavior about cancel during
```
Comment (mosabua) [inline _episodes/16.md:59]: Property to limit

File: _episodes/16.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+## Release 356
+
+Release notes discussed: <https://trino.io/docs/current/release/release-356.html>
+
+ - General:
+   - MATCH_RECOGNIZE - query over range of record with trending and such
+   - soundex function
+   - Property limit planning time (and improved behavior about cancel during 
+   planning)
+   - A bunch of performance improvements are pushdown (and start of docs for
```
Comment (mosabua) [inline _episodes/16.md:61]: around pushdown

File: _episodes/16.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+<iframe src="//www.slideshare.net/slideshow/embed_code/key/1fKHCGSRJwUjB7"  
+width="800" height="650" frameborder="0" marginwidth="0" marginheight="0" 
+scrolling="no" style="border:1px solid #CCC; border-width:1px; 
+margin-bottom:5px; max-width: 100%;" allowfullscreen> 
+</iframe>
+</p>
+
+### Druid Architecture
+
+Druid has several process types, briefly described below:
```
Comment (mosabua) [inline _episodes/16.md:91]: process types:

File: _episodes/16.md [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+ - **Router** processes are optional processes that can route requests to Brokers, Coordinators, and Overlords.
+ - **Historical** processes store queryable data.
+ - **MiddleManager** processes are responsible for ingesting data.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/16/druid-architecture.png"/><br/>
+The Druid architecture.
+</p>
+
+Druid processes can be deployed any way you like, but for ease of deployment we suggest organizing them into three server types: Master, Query, and Data.
```
Comment (mosabua) [inline _episodes/16.md:105]: wrap

File: _episodes/16.md [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+ - Data: Runs Historical and MiddleManager processes, executes ingestion workloads and stores all queryable data.
+ 
+Source: <https://druid.apache.org/docs/latest/design/architecture.html>.
+
+## PR of the week: PR 3522 Add Druid connector
+
+Our guest, Samarth, is the author of this weeks 
+[PR of the week](https://github.com/trinodb/trino/pull/3522). 
+[Puneet Jaiswal](https://twitter.com/puneetjaiswal) is the first engineer that
+started work to add a Druid connector. Later Samarth picked up the torch and The
```
Comment (mosabua) [inline _episodes/16.md:118]: the

File: _episodes/16.md [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+aggregation pushdown in the SPI. This enabled the use of the Druid connector to
+actually scale well with the completion of PR 4313 (see future work below).
+
+A [third honorable PR](https://github.com/trinodb/trino/pull/3881), 
+that was completed by [@findepi](https://twitter.com/findepi), was adding 
+pushdown to the jdbc client which appeared in release 337 along with the Druid 
+connector.
+
+It is incredible to see the amount of hands that various features and connectors
+pass through to get to the final.
```
Comment (mosabua) [inline _episodes/16.md:133]: the final release.

or 

the final state.

File: _episodes/16.md [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+
+
+### Future work:
+
+ - [(Recently Merged) SPI and optimizer rule for connectors that can support complete topN (PR 4249)](https://github.com/trinodb/trino/pull/4249)
```
Comment (mosabua) [inline _episodes/16.md:138]: Remove "Recently Merge)

**Review by mosabua — CHANGES_REQUESTED** [review]
A bunch of tweaks but then good to go

**Review by mosabua — APPROVED** [review]
Looks good now

**Pull request merged**